libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1044,8 +1049,11 @@ void AP_OSD_Screen::draw_speed_vector(uint8_t x, uint8_t y,Vector2f v, int32_t y
         int32_t interval = 36000 / SYM_ARROW_COUNT;
         arrow = SYM_ARROW_START + ((angle + interval / 2) / interval) % SYM_ARROW_COUNT;
     }
-
+    if (check_option(AP_OSD::OPTION_EXPANDED_DISPLAY)) {
+        backend->write(x, y, false, "%c%3.1f%c", arrow, u_scale(SPEED, v_length), u_icon(SPEED)); 
+    } else {
     backend->write(x, y, false, "%c%3d%c", arrow, (int)u_scale(SPEED, v_length), u_icon(SPEED));
```

Comment:
Formatting... It actually would be nice to format  whole ap_osd with ardupilot-astyle.sh.

Reply:
fixed...thanks
I will put doing that on my list....no problem...